### PR TITLE
Support more fractions

### PR DESCRIPTION
**Pull request by ronkok**
This PR adds support for `\genfrac`, `\brace`, `\brack`, `\above`, `\atopwithdelims`, `\overwithdelims`, and `\abovewithdelims`.

~~It deletes file `genfrac.js` and returns `groupTypes.genfrac` to files `buildHTML.js` and `buildMathML.js`. I did not want to do that, but I had to expose `groupTypes.genfrac` to several functions that have varying numbers of arguments.~~

### CODE REVIEW

**Comment by ylemkimon** [comment]
Can’t this be done like `[functions/accent.js](https://github.com/Khan/KaTeX/blob/master/src/functions/accent.js)`?

**Comment by ronkok** [comment]
@ylemkimon That might work. I'll look into it tomorrow.

**Comment by ronkok** [comment]
@ylemkimon I've followed your suggestion and placed the code into file `genfrac.js`. Thank you.

@kevinbarabash I've squashed the commits, or at least I've tried. Let me know if you want any further rearrangement.

**Comment by kevinbarabash** [comment]
@ronkok I'm going to hold off on generating screenshots for this until #1143 is merged.

**Comment by ylemkimon** [comment]
**Firefox**
binomtest-firefox-diff
[BinomTest-firefox.png](https://33-108306065-gh.circle-artifacts.com/0/image/BinomTest-firefox.png)

fractiontest-firefox-diff
[FractionTest-firefox.png](https://33-108306065-gh.circle-artifacts.com/0/image/FractionTest-firefox.png)


**Chrome**
binomtest-chrome-diff
[BinomTest-chrome.png](https://34-108306065-gh.circle-artifacts.com/0/image/BinomTest-chrome.png)

fractiontest-chrome-diff
[FractionTest-chrome.png](https://34-108306065-gh.circle-artifacts.com/0/image/FractionTest-chrome.png)

**Comment by kevinbarabash** [comment]
@ronkok any idea of why some of the fraction bars are wider?

**Comment by ronkok** [comment]
Yes, this PR supports `\genfrac` and `\abovewithdelims`, both of which give the author control over the fraction bar thickness. So I wrote screenshotter tests to demonstrate that capability.

**Comment by kevinbarabash** [comment]
> Yes, this PR supports \genfrac and \abovewithdelims, both of which give the author control over the fraction bar thickness. So I wrote screenshotter tests to demonstrate that capability.

Cool!  Sorry... I haven't actually look at the code yet.

File: src/functions/genfrac.js [hunk 1]
```diff
@@ -7,13 +9,267 @@ import Style from "../Style";
...
+            hasBarLine = true;
+            barSize = {
+                number: num,
+                unit: str.replace(/[+\-\d. ]/g, ''),
+            };
+        }
+    }
+
+    return [hasBarLine, barSize];
+};
```
Comment (ylemkimon) [inline src/functions/genfrac.js]: How about creating new argument type, e.g., `sizeOrEmpty`?

**Comment by ronkok** [comment]
@kevinbarabash, If you plan to do some PR reviews, I think that PR #1249 should take precedence over this one. After #1249 is merged, I'll circle back to this one and deal with the merge conflicts.

**Comment by ronkok** [comment]
I'm closing this PR temporarily. I'll reopen after I've dealt with a number of merge conflicts.

**Pull request closed without merging**